src/config.rs

```diff
@@ -70,6 +74,7 @@ impl Default for CrunchyConfiguration {
             ips_config: IPSConfiguration::default(),
             geoip_config: GeoIPConfiguration::default(),
             network_type_filter: None,
+            num_threads: DEFAULT_NUM_THREADS,
```

Comment:
What do you think of the idea of using `num_cpus::get()` or `std:thread:available_parallelism` to obtain, perhaps, a better default number-of-threads?  This wasn't part of original task, but it might be the right thing to do.  This had been my original thought for `spectre`, but I've amended my thinking on that:  I think it's good that the caller has control over how many threads are used.
Btw, the changes look great! 👍

Reply:
> but I've amended my thinking on that: I think it's good that the caller has control over how many threads are used.

This is how I think here: I would probably always use the max number of threads without thinking much - but I'm wondering, when would one decide to use less than max threads?

I don't mind either option tbh, I'm just thinking at loud

Reply:
When eg. you're running crunchy locally and would like to do something more than just looking how crunchy consumes your CPU 😄